### ISSUE TITLE
a11y: auto assign aria-label when using loading attribute

### ISSUE THREAD

**Issue by blackfalcon** [AlaskaAirlines/auro-button#298]
### Feature Description

When reviewing for accessibility with the auro-button, I came across the following issue with the loading state version of the button. See the attached screen shot. 

![Image](https://github.com/user-attachments/assets/4fd36d85-400b-41a0-981e-e17ab4c2955f)

When the auro-button uses the loading state, there are only the dancing dots, which is an SVG hidden from screens. Raising the issue of missing an accessible name. 

The feature request is to auto-fill the aria-label with `loading...` content when the `loading` attribute is used. 

Updating this line...

```js
aria-label="${ifDefined(this.arialabel ? this.arialabel : undefined)}"
```

to leverage the `loading` attribute

```js
aria-label="${ifDefined(this.loading ? 'Loading...' : (this.arialabel ? this.arialabel : undefined))}"
```

would easily address this feature. 

### Additional context

Currently, the loading state of the auro-button is failing a11y by default. This isn't a bug because a user could manually update the `arialabel` attribute on the auro-button when setting the loading state, but this is not documented to do so or commonly done, I would assume. 

Addign this auto feature would ensure that the loading state is always properly accessible.
